### DISCUSSION TITLE
What is the difference between nvim-dap-vscode-js and mason js adapters?

### DISCUSSION THREAD

**Discussion by Demianeen** [LazyVim/LazyVim#2170] [General]
I tried to switch from Tsserver to [Vtsls](https://github.com/yioneko/vtsls) ts LSP, so I needed to set up a DAP as well. When I was looking at the Lazyvim DAP [solution](https://www.lazyvim.org/extras/lang/typescript#nvim-dap-optional), I stumbled across js-debug-adapter usage that works only with node.js. I later found that I could set up a chrome-debug adapter with Mason as well. But I also decided to check what the outside world has to offer, and so I found [nvim-dap-vscode-js](https://github.com/mxsdev/nvim-dap-vscode-js).

I tried to find an article or something that would explain the difference between them but failed. Does anybody know what is the difference between them?

**Comment by davidfmatheson** [comment]
I'm trying to understand this, as well. From what I gather, LazyVim uses a plugin for integrating [Mason with DAP](https://www.lazyvim.org/extras/dap/core#mason-nvim-dapnvim). That is for all languages, to ensure that when you want to use a language, you will get the appropriate DAP installed. For example, for TypeScript it will install js-debug-adapter, which is the Mason name for [vscode-js-debug](https://github.com/microsoft/vscode-js-debug).

Once you have the right DAP for your language, you need to set up the adapter to tell nvim how to run the adapter when it wants to start debugging. Folks normally use [nvim-dap-vscode-js](https://github.com/mxsdev/nvim-dap-vscode-js) to set up the adapters for them, but LazyVim doesn't use this, it just sets up the adapters language-by-language (see [here](https://www.lazyvim.org/extras/lang/typescript#nvim-dap-optional) for the TypeScript setup). It seems like nvim-dap-vscode-js has [trouble with Mason](https://github.com/mxsdev/nvim-dap-vscode-js/issues/58) and maybe also the latest version of vscode-js-debug, which may be why LazyVim isn't using it.

If I let LazyVim do it's thing, and just override the launch configs a bit, I can get Node debugging working, tests and backend layers, but not the frontend debugging I'm used to doing with launching Chrome and setting breakpoints in my Angular code. I'm still fighting with that last part. My DAP config is [here](https://github.com/davidfmatheson/.dotfiles/blob/main/nvim-lazyvim/.config/nvim/lua/plugins/dap.lua).
